.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,52 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the
+{
+  "name": "Torchgeo DevContainer",
+  // Or use a Dockerfile or Docker Compose file. More info: https://containers.dev/guide/dockerfile
+  "image": "mcr.microsoft.com/devcontainers/python:3.10",
+
+  // Features to add to the dev container. More info: https://containers.dev/features.
+  "features": {
+    "ghcr.io/devcontainers/features/git:1": { "version": "os-provided" }
+  },
+
+  "postCreateCommand": "pip install .[datasets,docs,style,tests]",
+  "postStartCommand": "pip install -e .",
```

Comment:
Can these two be combined? Maybe we can use:
```
pip3 install -e .[all]
```
Not sure what the different between create and start are.

Reply:
`postCreateCommand` execute on the container creation when `postStartCommand ` execute on the container start. If you think the `.[datasets,docs,style,tests]` are likelly to change often then it's a way better practice to make a `pip3 install -e .[all]`.

To be honest I used to make this seperation back in the time, but it doesn't save much and I think `pip3 install -e .[all]` may be better